operator/maven-plugin/src/main/java/org/keycloak/operator/maven/ServerConfigGen.java

```diff
@@ -0,0 +1,67 @@
+package org.keycloak.operator.maven;
+
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.expr.StringLiteralExpr;
+import com.github.javaparser.utils.StringEscapeUtils;
+import org.apache.commons.text.CaseUtils;
+import org.apache.maven.plugin.logging.Log;
+import org.keycloak.config.AllOptions;
+import org.keycloak.config.Option;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class ServerConfigGen {
+
+    public static final String ANNOTATION_JSON_PROPERTY = "com.fasterxml.jackson.annotation.JsonProperty";
+    public static final String ANNOTATION_JSON_PROPERTY_DESCRIPTION = "com.fasterxml.jackson.annotation.JsonPropertyDescription";
+
+    public void generate(Log log, File destination) {
+        String className = "ServerConfig";
+        String[] packageName = new String[]{"org", "keycloak", "operator", "codegen", "configuration"};
+        CompilationUnit cu = new CompilationUnit();
+        cu.setPackageDeclaration(Arrays.stream(packageName).collect(Collectors.joining(".")));
+
+        ClassOrInterfaceDeclaration serverConfig = cu.addClass(className);
+
+        Path dest = Paths.get(destination.getAbsolutePath(), packageName);
+        dest.toFile().mkdirs();
+
+        AllOptions.ALL_OPTIONS.forEach(o -> {
+            if (o.getSupportedRuntimes().contains(Option.Runtime.OPERATOR)) {
+                String fieldName = CaseUtils.toCamelCase(o.getKey(), false, '-');
```

Comment:
do we really need to add the commons.text library when all we use is "toCamelCase" in this one place? I know, not reinventing the wheel, but looking at https://commons.apache.org/proper/commons-text/dependencies.html it can do a lot more and also has some transitive dependencies etc., so for this case I would vouch to implement this slightly trivial method ourselves and remove the dependency. But perhaps I am missing another point where it is used? Not exactly hard on that, and don't want to open up the productization discussion again, just saying if this is the only place it is used we should not introduce a dependency for this, as it is trivial to implement, most likely never changes and as such easy to maintain.

Reply:
laziness? Makes sense to have the implementation on ourselves, will update the PR with it 👍

Reply:
i'm totally for being lazy in 99.9% of cases. this is the 0.1% I am not ;)

Reply:
haven't found, rolled my own